src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -277,7 +283,15 @@ ReactDOMComponent.Mixin = {
   mountComponent: function(rootID, transaction, context) {
     this._rootNodeID = rootID;
 
-    assertValidProps(this, this._currentElement.props);
+    var props = this._currentElement.props;
+
+    var wrapper = wrappers[this._tag];
+    if (wrapper) {
+      wrapper.mountWrapper.call(this, props, context);
```

Comment:
Do you mean

```
switch (this._tag) {
case "input":
  ReactDOMInput.mountWrapper(this, props, context);
  props = ReactDOMInput.getNativeProps(this, props);
  break;
case "select":
  ReactDOMSelect.mountWrapper(this, props, context);
  props = ReactDOMSelect.getNativeProps(this, props);
  break;
}
```

or something else? If so, that seems like a lot of boilerplate. All the wrappers should have the same shape so I don't think it's polymorphic in this case?